lib/reflifc/include/reflifc/Declaration.h

```diff
@@ -112,14 +120,27 @@ namespace reflifc
         ifc::DeclSort sort() const { return index_.sort(); }
         ifc::DeclIndex index() const { return index_; }
 
+        Module declaring_module() const;
+
         bool operator==(Declaration other) const
         {
             assert(ifc_ == other.ifc_);
             return index_ == other.index_;
         }
 
+        friend std::hash<Declaration>;
+
     private:
         ifc::File const* ifc_;
         ifc::DeclIndex index_;
     };
 }
+
+namespace std
+{
+    template<>
+    struct hash<reflifc::Declaration>
+    {
+        size_t operator()(const reflifc::Declaration&) const;
```

Comment:
1. missing `noexcept` (http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-hash).
2. let's pass `Declaration` by value, it's small, trivially-copyable type.